app/src/main/java/org/mozilla/fenix/search/SearchFragmentStore.kt

```diff
@@ -308,7 +311,10 @@ private fun searchStateReducer(state: SearchFragmentState, action: SearchFragmen
         is SearchFragmentAction.AllowSearchSuggestionsInPrivateModePrompt ->
             state.copy(showSearchSuggestionsHint = action.show)
         is SearchFragmentAction.SetShowSearchSuggestions ->
-            state.copy(showSearchSuggestions = action.show)
+            state.copy(
+                searchSuggestionsEnabled = action.show,
+                showSearchSuggestions = action.show,
+            )
```

Comment:
Putting the changes here to the side, could we just do:

```suggestion
            is SearchFragmentAction.SearchDefaultEngineSelected ->
            state.copy(
                searchEngineSource = SearchEngineSource.Default(action.engine),
                showSearchSuggestions = action.settings.shouldShowSearchSuggestions,
                showSearchShortcuts = action.settings.shouldShowSearchShortcuts,
                showClipboardSuggestions = action.settings.shouldShowClipboardSuggestions,
                showHistorySuggestions = action.settings.shouldShowHistorySuggestions,
                showBookmarkSuggestions = action.settings.shouldShowBookmarkSuggestions,
                showSyncedTabsSuggestions = action.settings.shouldShowSyncedTabsSuggestions,
                showSessionSuggestions = true,
            )
```

I'm confused around the need for the new searchSuggestionsEnabled property as it seems we already have that info?

Reply:
We could, and that would fix this bug :). But by passing `Settings` in with the `Action` we are adding unnecessary complexity to this reducer and is bending the rule of trying to keep reducers "pure". This feels necessary because we're combining the state "Is this feature enabled" with "Is this feature currently visible" on the screen. By teasing it apart it will make the reducers simpler and easier to test at the cost of explicitly tracking both pieces of state.

Adding new properties to State doesn't always make it more complex, though I understand the desire to keep them small and concise. If we feel like that is necessary we should spend the time to improve or split up the model.

I think fixing it is out of the scope of this issue but I did want to chip away at it. 😊

Reply:
That all being said, happy to make the change if we don't want to start pulling on this thread now 😬

Reply:
Gotcha, thanks for the context! Appreciate the thought around this 👍 as we're already using `Settings` I think it may be best to be consistent (but not optimal) over having a mix of both approaches as the intention might not be clear for other maintainers (without commenting?) would another issue to refactor this be useful - that would remove any ambiguity over how the reducers should be used? May be worth getting others perspectives on this though.

Reply:
I'm good with that :) I'll think about how to approach revisiting these for refactoring

Reply:
Ok awesome, definitely appreciate the thought that has gone into making improvements